### PR TITLE
Add test cases for text box underline, strikethrough, and highlight features

### PR DESCRIPTION
**Pull request by joemoongit**
Had to make some modifications to lib/recipe/htmlToTextObjects.js and lib/recipe/text.js to make tests pass when 
```
html: true
```
is set as an option when invoking .text()

### CODE REVIEW

File: lib/recipe/htmlToTextObjects.js [hunk 1]
```diff
@@ -82,19 +82,22 @@ function parseNode(node) {
   const parsedData = {
     value,
     tag: node.tagName,
+    font: options ? options.font : undefined,
```
Comment (julianhille) [inline lib/recipe/htmlToTextObjects.js]: Are these options in the typescript definitions?

**Comment by julianhille** [comment]
Thank you very much for the pr

File: lib/recipe/htmlToTextObjects.js (hunk 1 above)
Comment (joemoongit) [reply lib/recipe/htmlToTextObjects.js]: These options are not in the typescript definitions. They are similar to the objects that are returned from the function _makeTextObject in lib/receipt/text.js

**Comment by julianhille** [comment]
please run prettier on the changed files, prettier is complaining about not compliant files.

see: https://github.com/julianhille/MuhammaraJS/actions/runs/5426565804/jobs/9942641353?pr=311

thanks

**Comment by joemoongit** [comment]
one item to note, when the option
```
html: true
```
is set, text wrapped in html tags seem to be created on a new line. i still need to look into this. should i address the fix in this pr? or create another one for it?

**Comment by julianhille** [comment]
i would suggest to start with it and then see. i'd use this as a base branch and then go from there, if its a small change related to this changes add it as commits, if its a complete seperate issue create a PR against this one.

sounds reasonable?

**Comment by julianhille** [comment]
I changed base too see if this runs in fixed ci. Would you mind reading on that prepare V4 branch?

**Comment by joemoongit** [comment]
i don't mind reading from that branch. i see it was deleted though. which branch should i direct this pr to?

**Comment by julianhille** [comment]
Ah. Dang. Sorry. Please reopen against dev.

**Comment by julianhille** [comment]
Test to reopen.

**Comment by joemoongit** [comment]
I created a commit to address the text wrapped in html tags appearing on a new line. I've also added a test case for bolded text. There are some spacing issues to address now.


[after_Add text with underline inside textbox.pdf](https://github.com/julianhille/MuhammaraJS/files/12247262/after_Add.text.with.underline.inside.textbox.pdf)
[before_Add text with underline inside textbox.pdf](https://github.com/julianhille/MuhammaraJS/files/12247263/before_Add.text.with.underline.inside.textbox.pdf)

File: lib/recipe/text.js [hunk 2]
```diff
@@ -97,6 +97,7 @@ function _initOptions(self, x = {}, y, options = {}) {
     self._firstLineHeight = 0; // indicates not set yet, determined later.
     self._textOptions = { textBox: {} };
     self._previousTextObjects = [];
+    self._previousTextObjects2 = [];
```
Comment (julianhille) [inline lib/recipe/text.js]: may be there a name which tells a bit more what this variables holds?

**Comment by julianhille** [comment]
sorry this branch has conflicts in the changelog, please rebase / merge dev

**Comment by joemoongit** [comment]
rebased. will work on spacing and may modify the changes a bit more

**Comment by joemoongit** [comment]
I will move the changes for the bug fix into its own branch and just keep the tests on this one

**Comment by julianhille** [comment]
linting has issues

**Comment by joemoongit** [comment]
I ran prettier on this diff, and it says that all files are using the prettier code style. Is there something else I have to do? I am using node v18.17

**Comment by julianhille** [comment]
against what folder? Lint action says: tests/recipe/text.js is needs prettier run

**Comment by joemoongit** [comment]
Ran it on the project's root folder

**Comment by julianhille** [comment]
did you compare versions of prettier?

**Comment by joemoongit** [comment]
ok, that was the issue. it looks like prettier was upgraded to a new major version. one thing though, now that i run prettier, its modifying many more files

**Comment by julianhille** [comment]
I'd suggest to just run it on the one file. Will add a PR fixing all the issues. Otherwise I might have a huge conflict.

**Comment by joemoongit** [comment]
ok, only ran it on the files that were modified on this pr

**Comment by julianhille** [comment]
last thing, i hope, would you mind adding a changelog entry or do you want me to?

- Added tests for yada yada :P

**Comment by joemoongit** [comment]
i added them under unreleased :smile:

**Comment by julianhille** [comment]
> i added them under unreleased 😄

im sorry just overlooked it. nevermind. ci run incoming....

**Comment by joemoongit** [comment]
Commit [65159af](https://github.com/julianhille/MuhammaraJS/pull/311/commits/65159af9c4af4535c726b230488eeeedfc892bef) was run with prettier and node v18.17/npm v9.6.7

Just made a new commit [d216ae7](https://github.com/julianhille/MuhammaraJS/pull/311/commits/d216ae749a9b1510535a5be380d9074c9a5f6c42) running prettier with node v20.17/npm v10.8.2

**Comment by julianhille** [comment]
🥳 finally a lint run :P

**Comment by julianhille** [comment]
One last thing. Please squash merge fixup some of the last commits and how you'd think it's good readable.

**Comment by joemoongit** [comment]
Squashed

**Review by julianhille — APPROVED** [review]

**Pull request merged**